### ISSUE TITLE
As a Workshop Member I would like to post literary pieces in order to receive critiques from others

### ISSUE THREAD

**Issue by victorkane** [victorkane/lit-drupal-lean#8]
Cloned from: 6: User Story Template
https://github.com/victorkane/lit-drupal-lean/issues/issue/6
### Problem Statement List

_Check or strikethrough if confirmed IRL as problems_
- [x] Email is a pain as a tool for a literary workshop
- [x] Someone has to manage the list membership
- [x] Too much stuff, impossible to filter if the group is big
- [x] Can't group individual pieces by author, date, etc.
- [x] Can't easily categorize
- [x] Work gets lost or misplaced
- [x] Hard to know whether/how to start getting published
- [x] Resources are "somewhere else"
### Prioritized Assumptions List
- [ ] We will be able to create a vibrant online community of writers wanting to share literary pieces and critique them
- [ ] Such an online community will be a much better experience than current email list based workshops
### Hypothesis Statements

_Drupal Lean Process uses the following atomic hypothesis statement (often called sub-hypothesis when they are created as a result of dividing up an unwieldy, difficult to test hypothesis into atomic chunks) which goes together really well with the User Story testable/implementable atom:_

_We believe that [doing this/ building this feature/ creating this experience] for [these people/ personas] will achieve [this outcome]. We will know this is true when we see [this market feedback, quantitative measure, or qualitative insight]._ _(taken from Gothelf, Jeff (2013-02-22). Lean UX: Applying Lean Principles to Improve User Experience, Ch 3)_

_Check or strikethrough if confirmed IRL as a worthwhile hypothesis for MVP building and testing_
- [ ] We believe that creating an online literary community for writers and publishers will create a vibrant community and create many opportunities for all. We will know this is true when we see a large number of sign-ups, a large number of posts from those that sign up, signficant activity around the posts (critiques, etc.), and a large record of emerging opportunities (invitations to publish, writers reporting their work published, success stories being posted, etc.).
### Personas

_Personas are separate issues (probably created while working on this user story issue) since they may be re-used in user stories._
- [ ] Victor the writer
- [ ] James the publisher
### User Story Card

_This is the actual issue title_
### User Story Confirmation
- [ ] MVP 01
#### Upon accessing the front page anonymous user sees an option block with three clickable choices:
- What is Literary Workshop?
  - First option goes to static explanatory page followed by same option block
- Sign up
  - Second choice goes to registration page for **Workshop Members**
- Just try it out right now 
  - Third option logs them in with a choice of user **bard** or **muse**
- **Demo user** clicks on a post and can access post details
- **Demo user** can add demo critique (automatically gets added to critiqued text comments-style)
- **Anonymous user** can register as **Workshop member** from the sign-up block
  - Once registered **Workshop member** sees current posts made by other Workshop members
    - Registered **Workshop member**  can make posts, which appear on the home page upon login if desired.
- When **Workshop member** accesses their account page they see their own posts and critiques they have made and received on their work
### User Story Conversation

To be expressed in the Comments section of this issue

**Comment by victorkane** [comment]
Before starting **MVP 01** let's create a base feature **lit_base** that
- specifies **Text** and **Critique** content type
- specifies role **Workshop member**
- specifies permissions for **Workshop member** over **Text** and **Critique**

and then another feature **lit_zen** which depends on **lit_base** and specifies **Zen** as theme, or else just include it in lit_base (for now).

**Comment by victorkane** [comment]
You don't have to download features and upload them to the server when you first create them any more! Advanced features lets you write the feature initially to a specified directory. Afterwards, it's drush feature management as always :)  Or just use the GUI if you really want to.
Make sure web server can write to the specified directory:

```
lit@awebfactory:~/lit-dev/docroot$ mkdir sites/all/modules/features
lit@awebfactory:~/lit-dev/docroot$ sudo chown www-data sites/all/modules/features/
lit@awebfactory:~/lit-dev/docroot$ ls -l sites/all/modules/features/
Module Lit Base written to sites/all/modules/features/lit_base
```

**Comment by victorkane** [comment]
Before implementing MVP 01, let's add automatic nodetitles for both texts and critiques (automatically generating title and hiding the title field).

Text: [node:field_text_type]: [node:field-short-title] by [current-user:name] [current-date:custom:YmdHi]
Critique: CRIT by [current-user:name] [current-date:custom:YmdHi], of: [node:field-text-being-critiqued]

**Comment by victorkane** [comment]
When we run the install profile, feature lit_layout doesn't get properly reverted, the masquerade block is not assigned to the proper region with the status 1 and lit_layout feature is in overridden state without the masquerade block being enabled. 
Workaround is to go in and revert lit_layout either through gui or drush. Then the block appears.
**So after install, go ahead and do a `drush features-revert-all` command.**
Still need to find out why, but not now :)

```
$ drush features-revert-all
The following modules will be reverted: lit_layout
Do you really want to continue? (y/n): y
Do you really want to revert lit_layout.fe_block_settings? (y/n): y
Reverted lit_layout.fe_block_settings. 
```

Better (even though a little dangerous, but we know what we're doing, don't we :) ?):

```
$ drush -y features-revert-all
```

**Comment by victorkane** [comment]
So we've created some content for default or demo content authored by default users bard and muse. Now, should we use Node export with features integration or UUID features to add our demo content to the install profile? See discussion at https://drupal.org/node/1311428. Based on https://drupal.org/comment/6997038#comment-6997038 I am inspired to try UUID features first, to see if the year that has passed has made its usability acceptable, at least for our purposes.

**Comment by victorkane** [comment]
So we have a bug in that node_export isn't designed at all to retain demo authorship of nodes. UUID Features, on the other hand, seems to retain uid's, which, in this case is fine.
So let's simply disable Lit Demo Content feature and not have it included in install program (not mentioned, not present in features to be reverted list) and instead create a Lit Default Content feature based on UUID Features.

**Comment by victorkane** [comment]
The saga continues:
- We get a warning before visiting our site:

```
Notice: Undefined property: stdClass::$nid in DrupalDefaultEntityController->load() (line 216 of /home/lit/lit-dev/docroot/includes/entity.inc).
Notice: Trying to get property of non-object in _node_index_node() (line 2784 of /home/lit/lit-dev/docroot/modules/node/node.module).
Warning: Attempt to assign property of non-object in node_build_content() (line 1399 of /home/lit/lit-dev/docroot/modules/node/node.module).
Notice: Trying to get property of non-object in node_build_content() (line 1419 of /home/lit/lit-dev/docroot/modules/node/node.module).
```
- We don't get authorship maintained :(

Let's add some code of our own for that !!

**Comment by victorkane** [comment]
So this looks very complicated for now. A [patch](https://drupal.org/files/uuid_features-2052387-clone-export.patch)  is underway for the above error, see https://drupal.org/node/2052387 and very recent attempt to fix it and some other problems at https://drupal.org/comment/8247997#comment-8247997
So, let's go back to the Node Export version, and just leave the demo content for now with four or five poems written by Anonymous, and come back to this problem of exporting and importing uuid based users (because each user does have a uuid!) at some later time.

**Comment by victorkane** [comment]
So what we need to do now is simply create a nice set of demo texts, then work on MVP 01 so when you click on demo you get logged in as bard or muse and you get to critique the poems.

**Comment by victorkane** [comment]
Let's break down MVP 01 into tasks:
- [x] Place landing page options block on front page
- [x] Implement About link and page, which appears above the options block
- [x] Implement demo auto login (bard, muse, as demo users) and views (any demo user can see demo view, any workshop member can see other views) (via "demo" tag)
- [x] Once logged in for demo, demo view block is placed above option block. Demo user clicks on a post and can access post details
- [x] Implement Demo user added demo critique (automatically gets added to critiqued text comments-style)
    (actually any text has all its critiques added to it below the detail view)
- [x] Implement registration workflow for **Workshop member**
- [ ] Once logged in as registered **Workshop member** user can make posts, which appear on the home page upon login if marked by authors as public to all registered Workshop members.
- [ ] User sees current posts made by other Workshop members on home page (if marked )
- [ ] When Workshop member accesses their account page they see their own posts and critiques they have made and received on their work.

Use these statements ending in "for MVP 01 #8" in commits.

**Comment by victorkane** [comment]
Reported as [d.o. issue 2162909](https://drupal.org/node/2162909)

We first had to solve the bug about why the lit_layout feature wasn't getting properly reverted by lit.install during the installation process. We could see in the recent log messages that for each of the three feature reverts, an error message was getting generated:

```
Warning: Invalid argument supplied for foreach() in _features_restore() (line 936 of /home/lit/lit-test/docroot/sites/all/modules/contrib/features/features.module).
```

The code in lit.install generated by Profiler Builder was as follows:

``` php
  // revert features to ensure they are all installed
  $features = array(
    'lit_base',
    'lit_demo_content',
    'lit_demo_layout'
  );
  features_revert($features);
```

Changing that to the following reduced the number of errors from three to two (and got the block actually appearing upon install!):

``` php
  // revert features to ensure they are all installed
  $features = array(
    'lit_base',
    'lit_demo_content'
  );
  features_revert($features);
  features_revert(array('lit_layout' => array('fe_block_settings')));
```

I finally got rid of all errors and got perfect feature reverts by using the formula <pre>features_revert(array('module' => array('component')));</pre> in the following code (manually replacing the generated code):

``` php
  // revert features to ensure they are all installed
  $features = array(
    array('lit_base' => TRUE),
    array('lit_demo_content' => TRUE),
  );
  features_revert($features);
  features_revert(array('lit_layout' => array('fe_block_settings')));
```

In the Recent log messages, the errors replaced with the following:

```
features    12/26/2013 - 12:34  Revert completed for lit_layout / fe_block_settings.    Anonymous (not verified)    
features    12/26/2013 - 12:34  Reverting lit_layout / fe_block_settings.   Anonymous (not verified)    
features    12/26/2013 - 12:34  Rebuild completed for lit_demo_content / node_export... Anonymous (not verified)    
features    12/26/2013 - 12:33  Rebuilding lit_demo_content / node_export_features. Anonymous (not verified)    
system  12/26/2013 - 12:33  lit_demo_content module enabled.    Anonymous (not verified)    
system  12/26/2013 - 12:33  lit_demo_content module installed.  Anonymous (not verified)
```

**Comment by victorkane** [comment]
We need to create a module for the landing page options block. For this we use the module builder module scaffolding module as a drush plugin (see #19).
In order to see which hooks we need to use, we do:

```
lit@awebfactory:~/lit-dev/docroot$ drush mb-list | grep block
...
  Preset block: Add blocks to your module
    hook_block_info
    hook_block_configure
    hook_block_save
    hook_block_view
lit@awebfactory:~/lit-dev/docroot$ 
```

To see all options for the `drush mb` command (to actually create the module scaffolding), do:

```
lit@awebfactory:~/lit-dev/docroot$ drush help mb
```

To create our application custom blocks module, we do:

```
lit@awebfactory:~/lit-dev/docroot$ drush mb lit_blocks block_info 
Enter the required hook preset groups (optional): 
Enter the required router paths, eg &#039;path/foo&#039; (optional): 
Enter the human readable name [Lit blocks]: ^C
lit@awebfactory:~/lit-dev/docroot$ drush mb lit_blocks block_info block_configure block_save block_view
Enter the required hook preset groups (optional): block
Enter the required router paths, eg &#039;path/foo&#039; (optional): sites/all/modules/custom
Enter the human readable name [Lit blocks]: L^H^H^H^H^H^H
Enter the description (optional) [TODO: Description of module]: 
Enter the help text (optional): 
Enter the dependencies (optional): 
Enter the package (optional): Lit
---- module folder: sites/all/modules/custom/lit_blocks
```

**Comment by victorkane** [comment]
Now that we have the scaffolding, a good resource for actually writing a custom block module (which we prefer in order to comply with the principle of everything in code) is [Four Kitchen's Building Custom Blocks with Drupal 7](http://fourkitchens.com/blog/2012/07/18/building-custom-blocks-drupal-7).
To understand this, one has to understand Drupal 7 Render Arrays, and why they even exist. See:
- [Randy Fay's Drupal 7 Render Arrays](http://randyfay.com/node/79).
- [Render Array Handbook Page on Drupal.org](http://drupal.org/node/930760)
- [Moshe Weitzman's Drupalcon San Francisco Presentation](http://sf2010.drupal.org/conference/sessions/page-render-drill-down-drupal-7)
- [The drupal_render() code](http://api.drupal.org/api/function/drupal_render/7)
- The Render Example module in the [Examples Project](https://drupal.org/project/examples) is a great way to explore render arrays. It has demonstrations of their use and allows you to explore the actual arrays describing any page.
- [Module Development: Drupal 7 Renderable Arrays - Daily Dose of Drupal Episode 39](http://www.youtube.com/watch?v=zizupA-PWZY)
- [The Scary Render Array](http://cocoate.com/ddbook/scary-render-array)

**Comment by victorkane** [comment]
Added hook_theme() to lit_blocks module with scaffolding:

```
lit@awebfactory:~/lit-dev/docroot$ drush mb lit_blocks theme --add --write
Enter the required hook preset groups (optional): block
Enter the required router paths, eg &#039;path/foo&#039; (optional): 
Enter the human readable name [Lit blocks]: 
Enter the description (optional) [TODO: Description of module]: 
Enter the help text (optional): 
Enter the dependencies (optional): 
Enter the package (optional): 
---- module folder: /home/lit/lit-dev/docroot/sites/all/modules/custom/lit_blocks
... (code followed, but was written thanks to --write directive)
```

**Comment by victorkane** [comment]
Sass, see #30

**Comment by victorkane** [comment]
For many of the sections pointed to by the landing page options block, we need panels or some kind of page manager, which Drupal doesn't have. The alternatives are:
- Panels and page manager
- Context
- Display suite...
- Special template files on the theme level

Special template files could be cool, but we don't want to work on the theme level.
Panels would be first choice, but everyone wants something lighter (like the Revamped Block System and Layout Support [Backdrop CMS is working on for their version 1.0](http://handbook.backdropcms.org/versions/roadmap/)).
For this project, let's go with the Context module, with which I can declare certain blocks (which can load arbitrary content in arbitrary layout) for certain URLs.

In this way, we can set up lightweight "panels" for various URLs by associating a Section content type assigned to a footer menu and managed by Context module. This should be part of the layout feature.

**Comment by victorkane** [comment]
- [x] remove "is" from pathauto exclusion settings
- [x] set pathauto url pattern for section content type
- [x] create footer menu, 
- [x] enable footer menu block in page bottom region
- [x] Create about section and add it section to footer menu
- [x] Create specific About context based on content type AND path (add in any old block just to see if it works)
- [x] Remove context layouts module from any dependencies, we're not using it since it depends on which theme you are using for layouts to be recognized (imagine if it weren't !)
- [x] Add everything, including static section content, to layout feature and recreate

**Comment by victorkane** [comment]
- [x] Create sections content type template (via https://drupal.org/node/224333#theme_hook_suggestions_2, see also https://drupal.org/node/223440) to enable section titles and sub-titles variables
- [x] See also https://drupal.org/node/917548 especially to check on existence of node variable (only on node pages)
- [x] Modify context to create theme template variables for section title / sub-title
- [x] Update function

**Comment by victorkane** [comment]
Steps to implement Demo:
- [x] bard and muse should be demo users
- [x] demo content should be marked with taxonomy or something during install (by convention, set the sticky field! hook_node_save or whatever to ensure this happens/doesn't happen ever, will have to be implemented)
- [x] add install task to randomly assign bard and muse as authors of demo content
- [x] demo section has single click to do demo as bard or as muse
- [x] upon login, all demo content on front page, all personal content on profile (account) page 
- [x] upon viewing text of others, single click to critique
- [x] upon viewing text detail, critiques are listed below

postponed: critiques can have critiques

**Comment by victorkane** [comment]
To flag stuff as demo, let's use a field we never use, the sticky field!
So just modify the fix authorship code to set the sticky field, and our demo security will have the criteria of users of role demo will have access to the demo views including sticky nodes filter, and users of role workshop member will have access to the views where (not sticky is in the filter).

**Comment by victorkane** [comment]
To create the demo section:
- [x] Create the section content
- [x] Add it to the footer menu
- [x] Add link to third landing page options block
- [x] Create specific Demo context based on content type AND path (add in any old block just to see if it works)
- [x] Add everything, including static section content, to layout feature and recreate

If Full HTML text format is desired it can be selected when editing section.

**Comment by victorkane** [comment]
- [x] Delete "No front page content has been created yet" (See https://drupal.org/node/1049290)
  My own solution in an overridden `page.tpl.php` template:

``` php
      <a id="main-content"></a>
      <?php if (drupal_is_front_page()): ?>
      <?php
          unset($title_prefix);
          unset($title);
          unset($title_suffix);
          unset($page['content']);
        ?>
      <?php endif; ?>
```

#### For Lit Layout feature
- [x] Add links to automatic logins to demo users in Demo section

#### For Lit Demo feature

Add demo functionality to demo section:
- [x] Implement automatic logins with Lit utils module packaged in Lit Demo feature, which must be enabled after install profile has completed
- [x] Implement account listing of own texts
- [x] Implement front page listing of all community (non-private) texts (<front>)
- [x] Implement embedded view of associated critiques to full text node page
- [x] Implement link to add critique of your own (postponed: nested threads of critiques, we're not doing disqus today)

**Comment by victorkane** [comment]
Menu condition doesn't work for everything in code contexts since it hard codes 'nid' (the curse of Drupal):

``` php

-    'menu' => array(
 -      'values' => array(
 -        'node/21' => 'node/21',
 -      ),
 -    ),
```

**Comment by victorkane** [comment]
Scaffolding for Lit utils module in which to implement menu callbacks to log in demo users programmatically:

```
lit@awebfactory:~/lit-dev/docroot$ drush mb lit_utils menu --write
Enter the required hook preset groups (optional): menu
Enter the required router paths, eg &#039;path/foo&#039; (optional): sites/all/modules/custom
Enter the human readable name [Lit utils]: 
Enter the description (optional) [TODO: Description of module]: Utils for Literary Workshop web app
Enter the help text (optional): 
Enter the dependencies (optional): 
Enter the package (optional): Lit
---- module folder: sites/all/modules/custom/lit_utils
```

**Comment by victorkane** [comment]
We're having trouble with menu links, nids, etc., so let's get rid of the footer menu altogether!
We have a block, and if user is anonymous, we leave it there.
For now, Demo feature only enables Lit utils module (later it will contain views).

**Comment by victorkane** [comment]
Demo Feature must be enabled manually (reverted) after install profile has completed.

**Comment by victorkane** [comment]
In order to implement the front page listing of all community (non-private) texts, we get rid of not printing content on front page since we want to take advantage of the Front page view.

**Comment by victorkane** [comment]
Using Chaos Tools Bulk Export Module (bulk_export) to place demo view (and non-demo front page view for now) into code, instead of features.
- Enabled Bulk Export Module
- Went to Structure > Bulk exporter

It is possible to create a module for:
- Context
- Environment indicator
- File entity
- System
- Views

We click on views, and select the Front Page (frontpage) view. I entered "lit_views_demo" into the module name field, and clicked on the `Export` button.

The result are three blocks, with code for the following files:
- lit_views_demo.info
- lit_views_demo.module
- lit_views_demo.views_default.inc

We can create the lit_views package module and place the lit_views_demo module in it, and integrate it into the demo feature!

**Comment by victorkane** [comment]
After last commit, installed from scratch, enabled demo module, masqueraded as bard, front page shows perfect cycling of demo content.
Selected five pieces of content of type text from the admin content page and made them non-sticky, so admin and workshop member view works a charm on front page too when they are logged in.

**Comment by victorkane** [comment]
For account page listing of own texts (My texts), let's:
- [x] Add a table view which shows all texts authored by current user
- [x] Add a context adding a block display of that view to user account page
- [x] Export both to a module, enable the module and revert both the context and the view.

<del>\- [ ] Place module in a Lit User feature</del> Don't f**_ing need a feature!!! 
    *_Ctools Bulk Export exports several features at once!!!**
- [x] With a simple push and then pull to test env and drush enabling, should work on test (and f***ing does!)

**Comment by victorkane** [comment]
For the implementation of embedded view of associated critiques to full text node page, together with a link to add critique of your own (postponed: nested threads of critiques, we're not doing disqus today):

One plan is:
- https://drupal.org/project/eva
- https://drupal.org/project/nodereference_url

(just like in Eaton's old D6 Views Gallery). However, the latter uses the References module (specifically node reference) instead of entity reference.

For an entity reference replacement, see discussion at https://drupal.org/node/1263118 and how far the solutions offered might take us. Actually, this discussion seems to have born a child: https://drupal.org/project/entityreference_prepopulate 

We should try out separate dummy entityreference_prepopulate and eva dry runs.

**Comment by victorkane** [comment]
We start with Entityreference prepopulate.
[Video Drupal 7 Entity Reference Prepopulate Module - Daily Dose of Drupal](http://www.youtube.com/watch?v=z9C4Uf6euLY)

Steps for using it in order to be able to add a way of easily adding a Critique to a Text:
- Enable entityreference_prepopulate
- Go to Critique content type, the one with the Text being critiqued entity reference field, and edit that field
- In the newly supported Additional Behaviors section, select the Entity reference prepopulate (Prepopulate entity reference values from URL) checkbox option.
- Immediately there appear three drop down choices:
  - Action (Action to take when prepopulating field with values via URL)
    - We select Disable field
      - Additional option: Apply action on edit (Apply action when editing an existing entity)
      - We check that too, since the enforcing relationship is intrinsic to the existing of the Critique
  - Fallback behavior
    - Determine what should happen if no values are provided via URL
    - We'll leave that at the default Do nothing selection
  - Skip access permission (Set a permission that will not be affected by the fallback behavior)
  - Additionally, support is present for integration with the Organic Groups module.
- Click the `Save settings` button.
- Since we left the Fallback behavior at Do nothing, we can still just go ahead and add a node instance of type Critique if we wish via the Add content option as usual, if desired.
- However, if we want to automatically add a critique to node/12 (a Text), we can invoke the following URL:
  - http://example.com/node/add/critique?field_text_being_critiqued=12
  - The add critique form appears, but with the Text being critiqued field greying out the prepopulated title and nid of the specified node! Awesome!
  - We could of course have selected other actions
    - Do nothing, leaving the field freely editable with the autocomplete widget
    - Hide field, in which case the submitted node would have the entity reference specified without the field actually being visible at all in the field.

The question now remains of how to create a link which appears every time we see a Text node (or reference to it, etc.), since the link is not supported itself out of the box. But that can be added in, using any method affecting the display of the node. Possible avenues:
- Panels module + tokens
- Display suite
- [Rules Link module](https://drupal.org/project/rules_link)
- Create a node template for the Critique content type, and easily render the link

We'll go ahead and create a content type specific node template.

Since the content type is already part of the Lit Base module, we will add to that feature the entity reference prepopulate module as well; committing the template implemented link at the same time.

**Comment by victorkane** [comment]
For the implementation of an embedded view of associated critiques for full text node pages, we will use the [EVA: Entity Views Attachment](https://drupal.org/project/eva) module.

But wait, let's KISS
- [x] Add some critiques to demo content
- [x] Create a view of critiques by text being critiqued
  - by text as contextual filter
  - take nid argument out of url if no results
- [x] Export view to module and place as dependency of Lit Base
- [x] Embed the view in text node template for full view only (create critique also)

This is not a particularly demo thing, this is a base thing, since it is intrinsic to any use of the text full node display. So the views module should go in as a dependency of Lit Base feature, where Text itself and the add a critique link is a dependency.

**Comment by victorkane** [comment]
Now that the Demo feature is initially complete, to close this User Story we need to satisfy the following:
- [ ] Once logged in as registered Workshop member user can make posts, which appear on the home page upon login <del>if marked by authors as public</del> to all registered Workshop members.
- [ ] User sees current posts made by other Workshop members on home page <del>(if marked public)</del>
- [ ] When Workshop member accesses their account page 
- They see their own posts. When a Text is displayed in full node view any critiques received for that Text are listed.
- They see all critiques they have made on the Texts of others

Almost everything has already been implemented. 

Acceptance test for gap analysis:
- Visit the site as Anonymous user
- Register as Workshop member "jamesj", "gertrudes", "josephh"
- Mail is received by them:

> lit@awebfactory.com.ar
> To gertrudes@awebfactory.com.ar
> Today at 3:56 PM
> gertrudes,
> 
> Thank you for registering at lit. Your application for an account is
> currently pending approval. Once it has been approved, you will receive
> another e-mail containing information about how to log in, set your password,
> and other details.
> 
> --  lit team

Admin receives a notification also:

> lit@awebfactory.com.ar
> To lit@awebfactory.com.ar
> Today at 11:37 PM
> jamesj has applied for an account.
> 
> http://lit-dev.opentimebook.com/user/14/edit
- Admin logs in and at People, edits user, assigns workshop member role and enables as active account

gertrudes receives mail:

> Account details for gertrudes at lit (approved)
> lit@awebfactory.com.ar
> To gertrudes@awebfactory.com.ar
> Today at 3:59 PM
> gertrudes,
> 
> Your account at lit has been activated.
> 
> You may now log in by clicking this link or copying and pasting it into your
> browser:
> 
> http://lit-dev.opentimebook.com/user/reset/12/1388861959/1L0i_No0NmteRnbHrHoAt3HsP30hjMaiyhfD_2EJuhE
> 
> This link can only be used once to log in and will lead you to a page where
> you can set your password.
> 
> After setting your password, you will be able to log in at
> http://lit-dev.opentimebook.com/user in the future using:
> 
> username: gertrudes
> password: Your password
> 
> --  lit team
- New workshop member logs in by clicking on the provided link, then editing and saving password and other user account details
- New workshop member creates four or five texts

<del>**Fail**</del>: user has no link on front page or anywhere else to create text. 
Fixed by commit b321265d757f55a51c169dbf57e84f88537a6ce3

<del>**Bug**</del>: what happened to the controls on the front page carousel?
Fixed by 8f0f00a1ef1bc18b38e9655ab5b11556b9b6e93c

<del>**Bug**</del> Workshop members and Demo users don't have permissions for text filters.
Fixed by 665731ad45647955e3c182c6a47e90c9b97dff0b
- Confirm that the texts are listed on the account page

Pass
- Repeat for another one or two users

Pass (gertrudes & jamesj)
- Confirm that all texts appear on the homepage for both workshop members

Pass
- Have each user make a critique on two or three of the others texts

Done
- Confirm that the critiques all show up on the full node display of the critiqued texts

Pass
- Confirm the all critiques made by a Workshop member appear listed on their account page

<del>**Fail**</del>

Fixed by 222b5c7cb168b4a647ee43fa9e0ba51829779c4b

**Comment by victorkane** [comment]
php.ini to prevent hanging upon user registration

```
realpath_cache_size = 512k
max_execution_time = 120
max_input_time = 120
memory_limit = 128M

# /etc/init.d/apache2 restart
```

For mail:

```
root@awebfactory:~# hostname opentimebook.com
root@awebfactory:~# echo opentimebook.com > /etc/hostname
```

Then installed postfix and sent some mails by hand:

```
lit@opentimebook:~$ sendmail victorkane@gmail.com
Subject: Keep trying
Never give up mind you
one
two
.
lit@opentimebook:~$ mail
-bash: mail: command not found
lit@opentimebook:~$ sendmail victorkane@gmail.com
subject: This is a test
from: lit@opentimebook.com
one
two
three
.
lit@opentimebook:~$ sendmail gertrudes@awebfactory.com.ar
subject: one
two
three
.
```

in another terminal:

```
tail -f /var/log/mail.log

Jan  4 18:45:48 opentimebook postfix/pickup[809]: 7CF5AC054A: uid=1001 from=<lit>
Jan  4 18:45:48 opentimebook postfix/cleanup[1367]: 7CF5AC054A: message-id=<20140104184548.7CF5AC054A@operntimebook.com>
Jan  4 18:45:48 opentimebook postfix/qmgr[810]: 7CF5AC054A: from=<lit@operntimebook.com>, size=291, nrcpt=1 (queue active)
Jan  4 18:45:51 opentimebook postfix/smtp[1369]: 7CF5AC054A: to=<victorkane@gmail.com>, relay=gmail-smtp-in.l.google.com[74.125.129.27]:25, delay=31, delays=29/0.01/0.27/2.4, dsn=2.0.0, status=sent (250 2.0.0 OK 1388861151 n8si49622817pax.131 - gsmtp)
Jan  4 18:45:51 opentimebook postfix/qmgr[810]: 7CF5AC054A: removed
Jan  4 18:47:15 opentimebook postfix/pickup[809]: E67DCC054A: uid=1001 from=<lit>
Jan  4 18:47:15 opentimebook postfix/cleanup[1367]: E67DCC054A: message-id=<20140104184715.E67DCC054A@operntimebook.com>
Jan  4 18:47:15 opentimebook postfix/qmgr[810]: E67DCC054A: from=<lit@operntimebook.com>, size=276, nrcpt=1 (queue active)
Jan  4 18:47:16 opentimebook postfix/smtp[1369]: E67DCC054A: to=<gertrudes@awebfactory.com.ar>, relay=awebfactory.com.ar[184.154.137.146]:25, delay=11, delays=10/0/0.22/0.27, dsn=2.0.0, status=sent (250 OK id=1VzWFg-0006ID-C7)
Jan  4 18:47:16 opentimebook postfix/qmgr[810]: E67DCC054A: removed
```
